stingray/lightcurve.py

```diff
@@ -664,38 +770,32 @@ def rebin(self, dt_new=None, f=None, method='sum'):
             raise ValueError("New time resolution must be larger than "
                              "old time resolution!")
 
-        if self.gti is None:
-
-            bin_time, bin_counts, bin_err, _ = \
-                utils.rebin_data(self.time, self.counts, dt_new,
-                                 yerr=self.counts_err, method=method)
-
-        else:
-            bin_time, bin_counts, bin_err = [], [], []
-            gti_new = []
-            for g in self.gti:
-                if g[1] - g[0] < dt_new:
-                    continue
-                else:
-                    # find start and end of GTI segment in data
-                    start_ind = self.time.searchsorted(g[0])
-                    end_ind = self.time.searchsorted(g[1])
+        bin_time, bin_counts, bin_err = [], [], []
+        gti_new = []
+        for g in self.gti:
+            if g[1] - g[0] < dt_new:
+                continue
+            else:
+                # find start and end of GTI segment in data
+                start_ind = self.time.searchsorted(g[0])
+                end_ind = self.time.searchsorted(g[1])
 
-                    t_temp = self.time[start_ind:end_ind]
-                    c_temp = self.counts[start_ind:end_ind]
-                    e_temp = self.counts_err[start_ind:end_ind]
+                t_temp = self.time[start_ind:end_ind]
+                c_temp = self.counts[start_ind:end_ind]
+                e_temp = self.counts_err[start_ind:end_ind]
 
-                    bin_t, bin_c, bin_e, _ = \
-                        utils.rebin_data(t_temp, c_temp, dt_new,
-                                         yerr=e_temp, method=method)
+                bin_t, bin_c, bin_e, _ = \
+                    utils.rebin_data(t_temp, c_temp, dt_new,
+                                     yerr=e_temp, method=method)
 
-                    bin_time.extend(bin_t)
-                    bin_counts.extend(bin_c)
-                    bin_err.extend(bin_e)
-                    gti_new.append(g)
+                bin_time.extend(bin_t)
+                bin_counts.extend(bin_c)
+                bin_err.extend(bin_e)
+                gti_new.append(g)
```

Comment:
It would be interesting to compare `extend` with `np.hstack` and see if one is faster than the other?

Reply:
```
In [2]: a = [1] * 100000000                                                                                                                                                                      

In [3]: b = [2] * 100000000                                                                                                                                                                      

In [4]: %time a.extend(b)                                                                                                                                                                        
CPU times: user 264 ms, sys: 172 ms, total: 436 ms
Wall time: 437 ms

In [7]: %time c = np.concatenate((a, b))                                                                                                                                                             
CPU times: user 15.8 s, sys: 1.41 s, total: 17.2 s
Wall time: 19 s

```

Reply:
👍